core/src/main/java/com/spotify/metrics/core/codahale/metrics/ext/Distribution.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2016 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.metrics.core.codahale.metrics.ext;
+
+import com.codahale.metrics.Counting;
+import com.codahale.metrics.Metric;
+
+import java.nio.ByteBuffer;
+
+public interface Distribution extends Metric, Counting {
+
+    /**
+     * Record value from Min.Double to Max.Double.
+     *
+     * @param val
+     */
+    void record(double val);
+
+    /**
+     * Return serialized distribution and flush.
+     *
+     * @return
+     */
+    ByteBuffer getValueAndFlush();
```

Comment:
Perhaps someone with more contributions to `semantic-metrics` can correct me if I am wrong, but I think that the core metrics types (Meter, Timer, etc) avoid having any knowledge of the serialized representation of the metric within them, and that is left up to "reporters" like the ffwd-reporter code. This sounds like it couples the serialized representation of a Distribution along with the data type. 

I recommend decoupling these two - lets say I was writing code to export `semantic-metrics` metrics to some other system besides ffwd, I might need to represent a Distribution differently on the wire for that system. One way to do this would be to change the return type here to be something like `Value` (it would probably make sense to avoid directly exposing types from the `t-digest` library

Reply:
Sorry, I think  my description in this context is confusing. I will fix that.

 So ByteBuffer is the distribution data point value. Serialization is done by the reporter.  

t-digest is never exposed that's why I added Distribution interface.